common/etc/nginx/include/s3gateway.js

```diff
@@ -352,7 +353,12 @@ function s3uri(r) {
         } else {
             path = basePath + uriPath;
         }
-    } else {
+    } else if (static_hosting) {
+        if (_isDirectory(uriPath)){
+            uriPath += "index.html";
```

Comment:
Please put the string literal `index.html` in a top level variable and reference it.